### PR TITLE
feat: add edit user teams

### PR DESCRIPTION
**Pull request by johnytiago**

### CODE REVIEW

File: packages/model/src/user.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export interface UserPatchRequest {
   skills?: string[];
   skillsDescription?: string;
   questions?: string[];
+  teams?: Omit<UserTeam, 'displayName' | 'role' | 'proposal'>[];
```
Comment (kmaid) [inline packages/model/src/user.ts:118]: Shouldn't this be a Pick to match the Joi Schema?

File: apps/asap-server/src/entities/user.ts [hunk 2]
```diff
@@ -75,6 +75,15 @@ export const userUpdateSchema = Joi.object({
...
   skillsDescription: Joi.string().allow(''),
   questions: Joi.array().items(Joi.object({ question: Joi.string() })),
+  teams: Joi.array().items(
+    Joi.object({
+      id: Joi.string().required(),
+      responsibilities: Joi.string(),
+      approach: Joi.string(),
+    })
+      .min(2)
+      .required(),
```
Comment (kmaid) [inline apps/asap-server/src/entities/user.ts:85]: Will this mean we always have to provide teams in a patch request?

**Review by kmaid — APPROVED** [review]

File: apps/asap-server/src/entities/user.ts (hunk 2 above)
Comment (johnytiago) [reply apps/asap-server/src/entities/user.ts:85]: No, just that the teams array has to have an object in it

File: packages/model/src/user.ts (hunk 1 above)
Comment (johnytiago) [reply packages/model/src/user.ts:118]: Was having a TS error using Pick, I will paste it here

Comment (kmaid) [reply packages/model/src/user.ts:118]: `teams?: Pick<UserTeam, 'id' | 'approach' | 'responsibilities'>[];`
seemed to work

**Pull request merged**